azurelinuxagent/common/logcollector.py

```diff
@@ -19,59 +19,83 @@
 
 import glob
 from heapq import heappush, heappop
+import logging
 import os
 import subprocess
 import zipfile
 
+from azurelinuxagent.common.conf import get_lib_dir, get_ext_log_dir
 from azurelinuxagent.common.future import ustr
-from azurelinuxagent.common.utils.fileutil import mkdir, read_file, rm_files, append_file, append_items_to_file
+from azurelinuxagent.common.utils.fileutil import mkdir, read_file, write_file, rm_files
 
+_EXTENSION_LOG_DIR = get_ext_log_dir()
+_AGENT_LIB_DIR = get_lib_dir()
 
-_LOG_COLLECTOR_DIR = '/var/lib/waagent/logcollector'
-_TRUNCATED_FILES_DIR = '/var/truncated'
+_LOG_COLLECTOR_DIR = os.path.join(_AGENT_LIB_DIR, "logcollector")
+_TRUNCATED_FILES_DIR = "/var/truncated"
```

Comment:
why keeping this outside the agent's directory?

Reply:
Moved to /var/lib/waagent/logcollector/truncated/